pdf-workitem/src/main/java/org/jbpm/process/workitem/pdf/GeneratePDFWorkitemHandler.java

```diff
@@ -0,0 +1,129 @@
+        displayName = "GeneratePDF",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.pdf.GeneratePDFWorkitemHandler()",
+        documentation = "${artifactId}/index.html",
+        parameters = {
+                @WidParameter(name = "TemplateXHTML", required = true),
+                @WidParameter(name = "PDFName")
+        },
+        results = {
+                @WidResult(name = "PDFDocument")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "${groupId}", artifact = "${artifactId}", version = "${version}")
+        },
+        serviceInfo = @WidService(category = "${name}", description = "${description}",
+                keywords = "pdf,generate,template,document,freemarker,xhtml",
+                action = @WidAction(title = "Generate PDF document")
+        ))
+public class GeneratePDFWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(GeneratePDFWorkitemHandler.class);
+    private static final String RESULTS_VALUE = "PDFDocument";
+    private static final String UTF_8 = "UTF-8";
+    private String resultXHTML;
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+
+        try {
+
+            RequiredParameterValidator.validate(this.getClass(),
+                                                workItem);
+
+            Map<String, Object> results = new HashMap<>();
+            String templateXHTML = (String) workItem.getParameter("TemplateXHTML");
+            String pdfName = (String) workItem.getParameter("PDFName");
+
+            if (pdfName == null || pdfName.isEmpty()) {
+                pdfName = "generatedpdf";
+            }
+
+            Configuration cfg = new Configuration(freemarker.template.Configuration.VERSION_2_3_26);
+            cfg.setDefaultEncoding("UTF-8");
+            cfg.setTemplateExceptionHandler(TemplateExceptionHandler.RETHROW_HANDLER);
+            cfg.setLogTemplateExceptions(false);
+
+            StringTemplateLoader stringLoader = new StringTemplateLoader();
+            stringLoader.putTemplate("pdfTemplate",
+                                     templateXHTML);
+            cfg.setTemplateLoader(stringLoader);
+
+            StringWriter stringWriter = new StringWriter();
+
+            Template pdfTemplate = cfg.getTemplate("pdfTemplate");
+            pdfTemplate.process(workItem.getParameters(),
+                                stringWriter);
+            resultXHTML = stringWriter.toString();
+
+            ITextRenderer renderer = new ITextRenderer();
+            renderer.setDocumentFromString(resultXHTML);
+            renderer.layout();
+
+            File generatedPDFile = File.createTempFile(pdfName,
+                                                       ".pdf",
+                                                       null);
+            FileOutputStream fos = new FileOutputStream(generatedPDFile);
```

Comment:
What about usin Document (jbpm-document) istead of a File? I thought you had that before....

Reply:
Once pdf is created almost always it will be sent to another handler like attach to gmail or upload to dropbox etc. I chose file because almost always those apis expect a File object and not DocumentImpl and then user would have to convert somewhere before. If this is not issue in your opinion then I'll change np.

Reply:
should we add to Document like methods toFile and fromFile?